src/WallpaperEngine/Input/Drivers/WaylandMouseInput.cpp

```
@@ -29,10 +76,62 @@ WallpaperEngine::Input::MouseClickStatus WaylandMouseInput::leftClick () const {
+
+    constexpr const char* request = "j/cursorpos";
+    if (send (fd, request, std::strlen (request), 0) < 0) {
+	close (fd);
+	return std::nullopt;
+    }
+    shutdown (fd, SHUT_WR);
+
+    std::string response;
+    char buffer[256];
+    ssize_t readBytes = 0;
+    while ((readBytes = recv (fd, buffer, sizeof (buffer), 0)) > 0) {
```

Cppcheck (2.20.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)